src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -222,6 +223,40 @@ export class TabsTitleControl extends TitleControl {
 		});
 	}
 
+	private updateTabSizing(): void {
+		if (this.tabsContainer) {
+			const options = this.accessor.partOptions;
+			if (options.tabSizing === 'fixed') {
+				this.tabsContainer.style.setProperty('--tab-sizing-max-width', options.tabSizingMaxWidth + 'px');
+				this.tabsContainer.onmouseenter = () => this.fixTabWidths();
```

Comment:
You need to use `addDisposableListener` and track disposal of the listeners.

Reply:
Yeah, didn't know that API, happy to use it.

Reply:
I may need guidance on this, please.

I used the `onmouseenter` and `onmouseleave` properties intentionally here so that the removal is trivial when the setting changes, and a new listener disposes of the one that may have been there previously (which it shouldn't).

This is OK, I believe, because this component – `TabsTitleControl` – owns the `tabsContainer` element.

Therefore, I don't think it's really necessary to use `addDisposableListener` because:
1. when `TabsTitleControl` is disposed, the DOM element is garbage-collected and with it the two listeners;
2. when the setting changes from fixed to some other tab sizing, the listeners are cleared out.

If we were adding listeners on somebody else's element, disposal would need to be much more explicit.

The situation might be simpler if we always put the two listeners there (which might happen in `registerTabsContainerListeners`) just like scrollbar listeners that are present even with wrapped tabs which don't scroll.

If I'm wrong in any of this, like I wrote above, I will need guidance, please.

Reply:
Commented in https://github.com/microsoft/vscode/pull/181729#discussion_r1193374514